File: Assets/Scripts/Game/Gameplay/Actors/MonsterTeleport.js
﻿//Monster Teleporting to move Monster to positions closer to the player if they're too far away.

public var Player : Transform;
public var Enemy : GameObject;
var DistanceNeededToTeleport : int;
var DistanceNeededToReactivate : int;
var Active : boolean;

function Update () {
	if(Player.GetComponent(UpgradeSystem).HasPlayerGottenUpgrade){
		if(Vector3.Magnitude(Player.position - this.transform.position) < DistanceNeededToTeleport && Active){
			navAgent = Enemy.GetComponent(NavMeshAgent);
			navAgent.enabled = false;
			Enemy.transform.position = this.transform.position;
			navAgent.enabled = true;
		
		
			Active = false;
		}
	}
	
	if(Vector3.Magnitude(Player.position - this.transform.position) > DistanceNeededToReactivate ){
		Active = true;
	}
	


}

File: Assets/Scripts/Game/Gameplay/Actors/MonsterTeleport.cs
using UnityEngine;
using System.Collections;

public class MonsterTeleport : MonoBehaviour {

	public Transform Player;
	public GameObject Enemy;
	public int DistanceNeededToTeleport;
	public int DistanceNeededToReactivate;
	public bool Active;
	public NavMeshAgent navAgent;
	
	// Update is called once per frame
	void Update () {

		if(Player.gameObject.GetComponent<UpgradeSystem>().HasPlayerGottenNextUpgrade){
			if(Vector3.Magnitude(Player.position - this.transform.position) < DistanceNeededToTeleport && Active){
				navAgent = Enemy.GetComponent<NavMeshAgent>();
				navAgent.enabled = false;
				Enemy.transform.position = this.transform.position;
				navAgent.enabled = true;
				
				Active = false;
			}
		}
		
		if(Vector3.Magnitude(Player.position - this.transform.position) > DistanceNeededToReactivate ){
			Active = true;
		}
	
	}
}
